testsuite/tests/polykinds/T9222.stderr

```diff
@@ -9,7 +9,7 @@ T9222.hs:14:3: error:
       ‘c’ is a rigid type variable bound by
         the type of the constructor ‘Want’:
           forall i1 j1 (a :: (i1, j1)) (b :: i1) (c :: j1).
-          ((a ~ '(b, c)) => Proxy b) -> Want a
+          ((a ~ '(b, c)) => Proxy b) ->@[n] Want a
```

Comment:
Ok to fix in another PR. But do you know what causes this? This is rather puzzling to me, as it doesn't look like what you did should have affected printing (or defaulting).

Reply:
Previously, if `-XLinearTypes` was off, GADTs defined using `->` used unrestricted arrow, and it appeared in the error message. After e019469f0e588385cab340163c7e1dcc8980a897, they use the linear arrow. Somewhere we generalize this to multiplicity polymorphism - I didn't check where, but this was already in the code.

Reply:
Interesting. It should have disappeared in the error message. This is an unusual error message, though, so maybe we need to do some magic switching in it.

Note that, also, `n` weirdly appears unbound. This is also interesting.

At any rate: make a task for it. Also, if it helps, this polymorphised type is generated by `dataConUserType` in `DataCon.hs`.